service/src/main/java/bio/terra/profile/service/spendreporting/azure/AzureSpendReportingService.java

```diff
@@ -147,4 +161,20 @@ private String getK8sNodeResourceGroupName(
         k8sResource.name(),
         k8sResource.regionName());
   }
+
+  @Scheduled(
```

Comment:
@sergiygetlin is it possible to introduce a TTL per cache entry rather than flushing the cache in bulk every N minutes? I'm envisioning a scenario where someone hits this API, the cache is populated then hits it again a minute later after the cache is flushed and we make another call to azure.

Reply:
@aherbst-broad There is a different way to remove items from cache like for instance "@CacheEvict" attribute, but it doesn't allow to setup TTL. Such thing as TTL should be set directly via Cache provider (Redis, etc.) since Spring provides only high-level cache abstraction to work with different cache solutions (at the same time it provides simple cache implementation such as ConcurrentHashMap). So, to use more sophisticated cache eviction policies we need to switch to non-default cache provider.